apps/benchmarks_keeper/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+pub mod api;
+pub mod config;
+pub mod price_update;
+pub mod types;
+pub mod utils;
+
+use alloy::providers::{Provider, ProviderBuilder, WsConnect};
+use api::fetch_price_data;
+use futures_util::StreamExt;
+use tracing::info;
+
+use crate::config::Config;
+use crate::price_update::PriceUpdate;
+use crate::utils::error::BenchmarksKeeperError;
+
+pub async fn run(config: Config) -> Result<(), BenchmarksKeeperError> {
+    info!("Starting Benchmarks Keeper");
+
+    // Create the WebSocket connection
+    let ws = WsConnect::new(&config.rpc_url);
+    println!("WebSocket connection created: {:?}", ws);
+    let provider = ProviderBuilder::new()
+        .on_ws(ws)
+        .await
+        .map_err(|e| BenchmarksKeeperError::RpcConnectionError(e.to_string()))?;
+
+    // Subscribe to PriceUpdate events
+    let filter = PriceUpdate::filter().address(config.contract_address);
+    let sub = provider.subscribe_logs(&filter).await?;
+
+    println!("Awaiting PriceUpdate events...");
+
+    let client = reqwest::Client::new();
+
+    // Process PriceUpdate events
+    let mut stream = sub.into_stream();
+    while let Some(log) = stream.next().await {
```

Comment:
We should probably have a reconnect on the error and a liveness check in case the streams gets stuck without producing new data.